src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const rows = sourceString.split(';').filter(Boolean);
+
+  for (const row of rows) {
+    const [key, value] = row.split(':');
+
+    if (value && key) {
+      const cleanedKey = key.trim();
+      const cleanedValue = value.trim();
+
+      result[cleanedKey] = cleanedValue;
+    }
+  }
```

Comment:
Instead of using a 'for...of' loop, consider using array iteration methods like 'forEach' to iterate over 'rows'. This aligns with the task's recommendation to avoid loops and use iteration methods.